### PR TITLE
Step/Pipeline configuration tests

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
This PR adds tests for the step and pipeline configuration as well as compilation.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

File: src/zenml/config/pipeline_configurations.py [hunk 1]
```diff
@@ -87,6 +89,18 @@ class PipelineRunConfiguration(StrictBaseModel):
...
     extra: Dict[str, Any] = {}
 
+    def yaml(self, **kwargs: Any) -> str:
+        """Yaml representation of the run configuration.
+
+        Args:
+            **kwargs: Kwargs to pass to the pydantic json(...) method.
+
+        Returns:
+            Yaml string representation of the run configuration.
```
Comment (strickvl) [inline src/zenml/config/pipeline_configurations.py:99]:
```suggestion
            Yaml string representation of the run configuration (with unsorted keys).
```

**Review by strickvl — COMMENTED** [review]
Looks pretty good. Just had some small comments here and there.

File: tests/unit/config/test_compiler.py [hunk 2]
```diff
@@ -11,3 +11,303 @@
...
+import pytest
+
+from zenml.config import ResourceSettings
+from zenml.config.base_settings import BaseSettings
+from zenml.config.compiler import Compiler
+from zenml.config.pipeline_configurations import PipelineRunConfiguration
+from zenml.config.step_configurations import StepConfigurationUpdate
+from zenml.exceptions import PipelineInterfaceError, StackValidationError
+from zenml.pipelines import pipeline
+
```
[the remark this answers is not in the document]
Comment (schustmi) [reply tests/unit/config/test_compiler.py:23]: Nice catch, that one was definitely missing. I added a test a for it

**Review by strickvl — APPROVED** [review]
Looks good now!

File: tests/unit/conftest.py [hunk 3]
```diff
@@ -221,6 +221,17 @@ def _step_function() -> None:
     return _generate_empty_steps
 
 
+@pytest.fixture
```
[the remark this answers is not in the document]
Comment (schustmi) [reply tests/unit/conftest.py:224]: I agree, it's really hard to keep track of all the fixtures we have to not duplicate/reimplement something by accident. However I think it's not a good idea to keep track of them in some other place, that will just be additional work and will be out of sync in a week or so. A quick google search gave me `pytest --fixtures` which gives us a list of all fixtures, I guess we can use that as long as our fixtures contain good docstrings/have expressive names?

**Pull request merged**